docs-src/self-hosted/data-encryption.md

```diff
@@ -28,30 +28,23 @@ Using encryption ensures that your sensitive data exists unencrypted only on the
 By default, your data is serialized to a [Payload](/concepts/what-is-a-payload) by a [Data Converter](/concepts/what-is-a-data-converter).
 To encrypt your Payload, configure your custom encryption logic with a [Payload Codec](/concepts/what-is-a-payload-codec) and set it with a [custom Data Converter](/concepts/what-is-a-custom-data-converter).
 
-A Payload Codec does byte-to-byte conversion to transform your Payload (for example, by implementing compression and/or encryption and decryption) and is an optional step that happens between the wire and the [Payload Converter](/concepts/what-is-a-payload-converter):
+A Payload Codec does byte-to-byte conversion to transform your Payload (for example, by implementing compression and/or encryption and decryption) and is an optional step that happens between the Client and the [Payload Converter](/concepts/what-is-a-payload-converter):
 
-```bash
-User code <--> Payload Converter <--> Payload Codec <--> Wire <--> Temporal Server
-```
+![Remote data encoding architecture](/diagrams/remote-data-encoding.svg)
 
-You can run your Payload Codec with a [Codec Server](/concepts/what-is-a-codec-server) and use the Codec Server endpoints in Web UI and tctl to decode your encrypted Payload locally.
+You can run your Payload Codec with a [Codec Server](/concepts/what-is-a-codec-server) and use the Codec Server endpoints in the Web UI and CLI to decode your encrypted Payload locally.
```

Comment:
super optional: Maybe CLI server? It's fine the way it is, it just feels like CLI is missing a noun.

Reply:
I'm going to keep this as-is because (unfortunately) we need to be a bit ambiguous about the CLI client as long as we're supporting two of them, as long as you don't feel more strongly about it!